go/app/model/item.go

```diff
@@ -0,0 +1,64 @@
+package model
+
+import (
+	"database/sql"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/google/uuid"
+)
+
+type Item struct {
+	Name string `json:"name"`
+	Category string `json:"category"`
+}
+
+type Items struct {
+	Items []Item `json:"items"` 
+}
+
+
+
+func GetItems(db *sql.DB) ([]Item, error) {
+	var err error
+	cmd := "SELECT * FROM items"
+	rows, _ := db.Query(cmd)
+	defer rows.Close()
+	//structを作成
+	var item_list []Item
+	//取得したデータをループでスライスに追加　for rows.Next()
+	for rows.Next() {
+		var item Item
+		var id uuid.UUID
+		//scan データ追加
+		err = rows.Scan(&id, &item.Name, &item.Category)
+		if err != nil {
+			return nil, err
+		}
+		item_list = append(item_list, item)
+	}
+	err = rows.Err()
+	if err != nil {
+		return nil, err
+	}
+	return item_list, nil
+}
+
+func AddItem(item Item, db *sql.DB) error {
+	id, err :=  uuid.NewUUID()
+	if err != nil {
+		return err
+	}
+	if db == nil {
+		return err
+	}
+	stmt, err := db.Prepare("INSERT INTO items (id, name, category) VALUES (?,?,?)")
```

Comment:
`curl -X POST http://127.0.0.1:9000/items -d name=jacket -d category=fashion`を実行すると、
```
{"message":"datatype mismatch"}
```
とレスポンスがあり、ログの方にも
```
{"time":"2022-05-19T14:47:20.262804+09:00","level":"ERROR","prefix":"echo","file":"main.go","line":"43","message":"datatype mismatch"}
```
とエラーログが出ました。

原因は、itemsテーブルのid（`integer`型）に`UUID`型([16]byte)を入れようとしているからだと思います。

Reply:
```suggestion
	stmt, err := db.Prepare("INSERT INTO items (name, category) VALUES (?,?)")
```

テーブル生成のクエリ文に、`primary key`を指定していると思うのですが、これはINSERTする際にデータが与えられなかったら自動的にidを代入してくれるので、データ入力を省略して良いと思います！